src/main/java/io/phdata/streamliner/schemadefiner/util/StreamlinerUtil.java

```diff
@@ -245,4 +258,108 @@ public static boolean deleteDirectory(File directoryToBeDeleted) {
     }
     return directoryToBeDeleted.delete();
   }
+
+  public static boolean fileExists(String configPath) {
+    return Files.exists(Paths.get(configPath));
+  }
+
+  public static Map<String, Map<String, String>> readTypeMappingFile(String path)
+      throws IOException {
+    if (!fileExists(path)) {
+      throw new FileNotFoundException("Type mapping file not found: " + path);
+    }
+    ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
+    Map<String, Map<String, String>> typeMappingFile = mapper.readValue(new File(path), Map.class);
+    return typeMappingFile;
+  }
+
+  public static List<File> listFilesInDir(String path) throws FileNotFoundException {
+    File d = new File(path);
+    if (d.exists() && d.isDirectory()) {
+      return Arrays.stream(d.listFiles())
+          .filter(file -> file.isFile())
+          .collect(Collectors.toList());
+    } else {
+      throw new FileNotFoundException("Directory path does not exist" + path);
+    }
+  }
+
+  public static void writeFile(String content, String fileName) {
+    log.info("Writing file: {}", fileName);
+    log.trace("File content: {}", content);
+
+    try (FileWriter fw = new FileWriter(fileName)) {
+      fw.write(content);
+    } catch (IOException e) {
+      log.error("Error writing file: {}", fileName);
+      throw new RuntimeException(e);
+    }
+  }
+
+  public static void isExecutable(String path) {
+    File file = new File(path);
+    log.debug("File: {}, executable flag: {}", path, file.canExecute());
+    if (file.canExecute()) {
+      setExecutable(path);
+    }
+  }
+
+  private static void setExecutable(String path) {
+    File f = new File(path);
+    f.setExecutable(true);
+  }
+
+  public static String readFile(String path) {
+    log.debug("Reading file: {}", path);
+    StringBuilder sb = new StringBuilder();
+    try (BufferedReader br = new BufferedReader(new FileReader(path))) {
+      while (br.ready()) {
+        sb.append(br.readLine());
+        sb.append("\n");
+      }
+    } catch (FileNotFoundException e) {
+      log.error("File not found : {}", path);
+      throw new RuntimeException(e);
+    } catch (IOException e) {
+      e.printStackTrace();
```

Comment:
If we want the stack trace, let's use the log not std err.

Reply:
yeah I missed to remove this. I will add log.error